src/main/java/org/mule/tools/devkit/sonar/checks/structure/GitIgnoreExistsCheck.java

```diff
@@ -30,16 +27,44 @@
 
     public static final String KEY = "gitignore-exist";
     private static final String PATH = ".gitignore";
-    private static final ImmutableList<String> REQUIRED_GITIGNORE_FIELDS = ImmutableList.of("*.class", "*.jar", "*.war", "target/", ".classpath", ".settings/", ".project",
-            ".factorypath", ".idea/", "*.iml", "*.ipr", "*.iws", "bin/", ".DS_Store", "automation-credentials.properties", "muleLicenseKey.lic");
-
     private static FileSystem fileSystem;
+    public Boolean applyByTrue = true;
+
+    public ImmutableList<String> requiredGitignoreFields = ImmutableList.of("\\*\\.class", "\\*\\.jar", "\\*\\.war", "[/]?target[/]?", "[*]?[/]?\\.classpath",
+            "[*]?[/]?\\.settings[/]?", "[*]?[/]?\\.project",
+            "[*]?[/]?\\.factorypath", "[/]?\\.idea[/]?", "\\*\\.iml", "\\*\\.ipr", "\\*\\.iws", "[/]?bin[/]?", "[/]?\\.DS_Store", "automation-credentials\\.properties",
+            "muleLicenseKey.lic");
+
+    public String issueMessage = ".gitignore file in project is missing the following exclusions: '%s'.";
 
     public GitIgnoreExistsCheck(FileSystem fileSystem) {
         super(fileSystem);
         this.fileSystem = fileSystem;
     }
 
+    public class matchesPredicate<T extends String> implements Predicate<T> {
+
+        private Iterable<String> target;
+        private Boolean applyByTrue;
+
+        public matchesPredicate(Collection<String> target, Boolean applyByTrue) {
+            this.target = target;
+            this.applyByTrue = applyByTrue;
+        }
+
+        @Override
+        public boolean apply(T t) {
+            for (String s : target) {
+                if (Pattern.compile(t)
+                        .matcher(s)
+                        .matches()) {
+                    return !applyByTrue;
+                }
+            }
+            return applyByTrue;
+        }
+    }
```

Comment:
Not sure I understand the purpose of this class (particularly the applyByTrue attribute).
Please explain (it looks to me that it can be replaced by Predicates.or() and multiple instances of Predicates.containsPattern()).

Also, the following:
 - Class names should ALWAYS start with upper case.
 - If it's used only in this class, then it should be private, not public. Otherwise it should be on a separated file.
 - Doing a return in the middle of an iteration such as a for is very similar to doing a break and can be improved.